src/modules/Builder/features/PerformanceTasks/Flanker/Flanker.test.tsx

```diff
@@ -0,0 +1,724 @@
+    expect(headCells).toHaveLength(4);
+    headCells.forEach((headCell, index) => {
+      expect(headCell).toHaveTextContent(`Block ${index + 1}`);
+    });
+
+    const bodyRows = screen.getByTestId(`${testId}-table`).querySelectorAll('tbody tr');
+    expect(bodyRows).toHaveLength(2);
+    const [firstRow, secondRow] = bodyRows;
+
+    expect(firstRow.querySelectorAll('td')).toHaveLength(4);
+    expect(secondRow.querySelectorAll('td')).toHaveLength(4);
+
+    firstRow.querySelectorAll('td').forEach((cell) => {
+      expect(cell).toHaveTextContent('text');
+    });
+    secondRow.querySelectorAll('td').forEach((cell) => {
+      expect(cell).toHaveTextContent('text');
+    });
+  });
+
+  describe('Validations', () => {
+    test.each`
+      testId                                                     | error                                   | description
+      ${`${mockedFlankerTestid}-buttons-0-text`}                 | ${'Button Name or Image is required'}   | ${'Left Button Name'}
+      ${`${mockedFlankerTestid}-buttons-1-text`}                 | ${'Button Name or Image is required'}   | ${'Right Button Name'}
+      ${`${mockedFlankerTestid}-stimulus-screen`}                | ${'Please fill in all required fields'} | ${'Stimulus Screen'}
+      ${`${mockedFlankerTestid}-practice-round-block-sequences`} | ${'Add stimulus screens first'}         | ${'Practice Round: Stimulus Screen'}
+      ${`${mockedFlankerTestid}-test-round-block-sequences`}     | ${'Add stimulus screens first'}         | ${'Test Round: Stimulus Screen'}
+    `('$description', async ({ testId, error }) => {
+      const ref = renderFlanker();
+
+      await ref.current.trigger('activities');
+
+      await waitFor(() => {
+        const section = screen.getByTestId(testId);
+
+        expect([...section.querySelectorAll('p')].at(-1)).toHaveTextContent(error);
+      });
+    });
+
+    test.each`
+      testId                                                             | attribute                          | error                                       | description
+      ${`${mockedFlankerTestid}-general-instruction-instruction`}        | ${'activities.0.items.0.question'} | ${'Overview Instruction is required'}       | ${'Empty Overview Instruction'}
+      ${`${mockedFlankerTestid}-practice-round-instruction-instruction`} | ${'activities.0.items.1.question'} | ${'Practice Round Instruction is required'} | ${'Empty Practice Round Instruction'}
+      ${`${mockedFlankerTestid}-test-round-instruction-instruction`}     | ${'activities.0.items.7.question'} | ${'Test Round Instruction is required'}     | ${'Empty Test Round Instruction'}
+    `('$description', async ({ testId, attribute, error }) => {
+      const ref = renderFlanker();
+
+      expandAllPanels();
+
+      ref.current.setValue(attribute, '');
+      await ref.current.trigger('activities');
+
+      await waitFor(() => {
+        const instruction = screen.getByTestId(testId);
```

Comment:
you can get rid of `await waitFor()` wrapper, if you will use `find*` method:
`const instruction = await screen.findByTestId(testId);`